types/office-runtime/index.d.ts

```diff
@@ -0,0 +1,155 @@
+       *
+       */
+      setItem(key: string, value: string, callback?: (error?: Error) => void): Promise<void>;
+      /**
+       * @beta
+       * Removes an item for specified key and invokes a callback when finished. Returns a Promise. 
+       *  
+       *
+       */
+      removeItem(key: string, callback?: (error?: Error) => void): Promise<void>;
+      /**
+       * @beta
+       * Erases all AsyncStorage. Returns a Promise. 
+       *
+       */
+      clear(callback?: (error?: Error) => void): Promise<void>;
+       /**
+       * @beta
+       * Returns a Promise with all keys for your custom function. 
+       *
+       */
+      getAllKeys(callback?: (error?: Error, keys?: string[]) => void): Promise<string[]>;
+       /**
+       * @beta
+       * Stores multiple key-value pairs in one batch. Returns a Promise. When a key doesn't exist, returns null. 
+       *
+       */
+      multiSet(keyValuePairs: string[][], callback?: (errors?: Error[]) => void): Promise<void>;
+       /**
+       * @beta
+       * Deletes a batch of keys, specified in the `keys` array.
+       *
+       */
+      multiRemove(keys: string[], callback?: (errors?: Error[]) => void): Promise<void>;/**
+      /**
+      * @beta
+      * Fetches a batch of keys, specified in the `keys` array. When a key doesn't exist, returns null. 
+      * 
+      */
+      multiGet(keys: string[], callback?: (errors?: Error[], result?: string[][]) => void): Promise<string[][]>;
+    }
+  
+    /**
+    * @beta
+    * Object representing the dialog box for Excel Custom Functions only. 
+    *
+    */
+    interface Dialog {
+       /**
+       * @beta
+       * Method that closes a dialog box. 
+       */
+      close(): Promise<void>;
+    }
+  
+       /**
+       *
+       * Provides options for how a dialog is displayed for Excel Custom Functions only. 
+       *
+       */
+    interface DisplayWebDialogOptions {
+      /**
+       * @beta
+       * Determines whether the dialog box displays as a popup (false) or within an IFrame (true). This setting is only applicable to custom functions running on Excel Online. 
+       *
+       * [Api set: Dialog 1.2]
+       */
+      displayInIFrame?: boolean;
+      /**
+       * @beta
+       * Defines the height of the dialog box as a percentage of the current display.
+       *
+       * [Api set: Dialog 1.2]
+       */
+      height?: string;
+  
+      /**
+       * @beta
+       * Defines the width of the dialog box as a percentage of the current display. 
+       *
+       * [Api set: Dialog 1.2]
+       */
+      width?: string;
+  
+      /**
+       * @beta
+       * True if title is hidden from the dialog box. 
+       *
+       * [Api set: Dialog 1.2]
+       */
+      hideTitle?: boolean;
+  
+      /**
+       * @beta
+       * Callback that is run when the dialog box sends a message to its parent.
+       *
+       *
+       */
+      onMessage?: (message: string, dialog?: Dialog) => void;
```

Comment:
the short form is preferred iirc eg. `onMessage?(message: string, dialog?: Dialog): void`